### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

File: src/neo-vm/Types/Integer.cs [hunk 1]
```diff
@@ -10,36 +10,32 @@ public class Integer : PrimitiveType
     {
         public const int MaxSize = 32;
 
-        private readonly int _length;
         private readonly BigInteger value;
 
+        public override ReadOnlyMemory<byte> Memory => value.IsZero ? ReadOnlyMemory<byte>.Empty : value.ToByteArray();
```
Comment (shargon) [inline src/neo-vm/Types/Integer.cs]: we can cache the result

File: src/neo-vm/Types/PrimitiveType.cs [hunk 2]
```diff
@@ -6,52 +6,53 @@ namespace Neo.VM.Types
...
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj is PrimitiveType p) return Equals(p);
+            return false;
         }
 
-        public virtual int GetByteLength()
+        public virtual bool Equals(PrimitiveType other)
         {
-            return ToByteArray().Length;
+            if (ReferenceEquals(this, other)) return true;
```
Comment (shargon) [inline src/neo-vm/Types/PrimitiveType.cs]: This line could be called twice. L15 and here

File: src/neo-vm/Types/Integer.cs (hunk 1 above)
Comment (erikzhang) [reply src/neo-vm/Types/Integer.cs]: I don't want to cache it because it will double the space.

File: src/neo-vm/Types/PrimitiveType.cs (hunk 2 above)
Comment (erikzhang) [reply src/neo-vm/Types/PrimitiveType.cs]: But this method may also be called separately.

**Pull request merged**